### PR TITLE
[jigson] New port

### PR DESCRIPTION
**Pull request by JoshuaSledden**
Jigson is a header-only C++ library that facilitates mapping between JSON data
and C++ class objects. The name is a play on "Jigsaw" and "JSON", reflecting
its purpose of piecing class objects together from JSON data.

Features:
- Header-only library for easy integration
- Depends on nlohmann-json for JSON parsing
- Supports C++20
- Provides intuitive mapping between JSON and C++ objects

This commit adds the initial port for Jigson version 0.1.2.

Example Usage:
```cpp
#include <iostream>
#include <jigson\Jigson.hpp>
using namespace jigson;

// An example class we want to map to
class TestClass {
public:
  std::string name;
  int value;
};

// Specialized mapping profile configuration.
template<>
inline void mapping::MapperFactory::configure_profile<TestClass>(MappingProfile<TestClass>& profile) {
  profile.add_mapping("name", [](TestClass& test_class, const json_payload& json) {
    test_class.name = json.get<std::string>();
  });

  profile.add_mapping("value", [](TestClass& test_class, const json_payload& json) {
    test_class.value = json.get<int>();
  });
}

int main() {
  // Register a mapping profile for a class
  mapping::register_profile<TestClass>();

  // Example JSON object (Can also parse a file)
  nlohmann::json json_payload = {
    {"name", "Hello, Jigson!"},
    {"value", 42}
  };

  // Map JSON to the class type
  const auto [name, value] = mapping::map<TestClass>(json_payload);

  std::cout << name << ". You are " << value << " Years Old." << std::endl;
  return 0;
}

```

- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [x] The name of the port matches an existing name for this component on https://repology.org/ if possible, and/or is strongly associated with that component on search engines.
- [x] Optional dependencies are resolved in exactly one way. For example, if the component is built with CMake, all `find_package` calls are REQUIRED, are satisfied by `vcpkg.json`'s declared dependencies, or disabled with [CMAKE_DISABLE_FIND_PACKAGE_Xxx](https://cmake.org/cmake/help/latest/variable/CMAKE_DISABLE_FIND_PACKAGE_PackageName.html).
- [x] The versioning scheme in `vcpkg.json` matches what upstream says.
- [x] The license declaration in `vcpkg.json` matches what upstream says.
- [x] The installed as the "copyright" file matches what upstream says.
- [x] The source code of the component installed comes from an authoritative source.
- [x] The generated "usage text" is accurate. See [adding-usage](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/examples/adding-usage.md) for context.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is in the new port's versions file.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

File: ports/jigson/usage [hunk 1]
```diff
@@ -0,0 +1,4 @@
+vcpkg-sample-library provides CMake targets:
+
+find_package(jigson CONFIG REQUIRED)
+target_link_libraries(main PRIVATE jigson::jigson)
```
Comment (dg0yt) [inline ports/jigson/usage:4]:
```suggestion
jigson provides CMake targets:

  find_package(jigson CONFIG REQUIRED)
  target_link_libraries(main PRIVATE jigson::jigson)
```

File: ports/jigson/jigson-config.cmake.in [hunk 2]
```diff
@@ -0,0 +1,6 @@
+if(NOT TARGET jigson::jigson)
+    add_library(jigson::jigson INTERFACE IMPORTED)
+    set_target_properties(jigson::jigson PROPERTIES
+        INTERFACE_INCLUDE_DIRECTORIES "${_VCPKG_INSTALLED_DIR}/${VCPKG_TARGET_TRIPLET}/include"
+    )
```
Comment (dg0yt) [inline ports/jigson/jigson-config.cmake.in:5]: IIUC nlohmann_json is a transitive usage requirement.
```suggestion
    include(CMakeFindDependencyMacro)
    find_dependency(nlohmann_json CONFIG)
    add_library(jigson::jigson INTERFACE IMPORTED)
    set_target_properties(jigson::jigson PROPERTIES
        INTERFACE_INCLUDE_DIRECTORIES "${_VCPKG_INSTALLED_DIR}/${VCPKG_TARGET_TRIPLET}/include"
        INTERFACE_LINK_LIBRARIES nlohmann_json::nlohmann_json
    )

```

File: ports/jigson/portfile.cmake [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+vcpkg_install_copyright(FILE_LIST "${SOURCE_PATH}/LICENSE")
+
+configure_file("${CMAKE_CURRENT_LIST_DIR}/jigson-config.cmake.in" "${CURRENT_PACKAGES_DIR}/share/${PORT}/jigson-config.cmake" @ONLY)
+
+# Copy usage examples
+file(INSTALL "${CMAKE_CURRENT_LIST_DIR}/usage"
+     DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}")
+
+set(VCPKG_POLICY_EMPTY_PACKAGE enabled)
```
Comment (JonLiu1993) [inline ports/jigson/portfile.cmake:21]:
```suggestion
set(VCPKG_POLICY_EMPTY_PACKAGE enabled)

```

File: ports/jigson/usage [hunk 4]
```diff
@@ -0,0 +1,4 @@
+jigson provides CMake targets:
+
+  find_package(jigson CONFIG REQUIRED)
+  target_link_libraries(main PRIVATE jigson::jigson)
```
Comment (JonLiu1993) [inline ports/jigson/usage:4]:
```suggestion
  target_link_libraries(main PRIVATE jigson::jigson)
  
```

**Review by JonLiu1993 — DISMISSED** [review]

**Comment by JonLiu1993** [comment]
Tested usage successfully by jigson:x64-windows:
````
jigson provides CMake targets:

  find_package(jigson CONFIG REQUIRED)
  target_link_libraries(main PRIVATE jigson::jigson)
````

**Review by BillyONeal — DISMISSED** [review]

File: ports/jigson/jigson-config.cmake.in [hunk 5]
```diff
@@ -0,0 +1,10 @@
+if(NOT TARGET jigson::jigson)
+    include(CMakeFindDependencyMacro)
+    find_dependency(nlohmann_json CONFIG)
+    add_library(jigson::jigson INTERFACE IMPORTED)
+    set_target_properties(jigson::jigson PROPERTIES
+        INTERFACE_INCLUDE_DIRECTORIES "${_VCPKG_INSTALLED_DIR}/${VCPKG_TARGET_TRIPLET}/include"
```
Comment (dg0yt) [inline ports/jigson/jigson-config.cmake.in:6]:
```suggestion
        INTERFACE_INCLUDE_DIRECTORIES "${_VCPKG_INSTALLED_DIR}/${VCPKG_TARGET_TRIPLET}/include/jigson"
```
Not tested.
Documentation says `#include <jigson.hpp>`, but installation goes to a subdir.

Comment (JoshuaSledden) [reply ports/jigson/jigson-config.cmake.in]: Good set of eyes :) thanks

**Review by JonLiu1993 — APPROVED** [review]

**Comment by BillyONeal** [comment]
> @JonLiu1993 anything else I need to do to get this merged in or am I waiting on a maintainer to merge?

I was going to merge it on Friday but the build wasn't done yet :/

Thanks for the new port!

**Comment by JoshuaSledden** [comment]
@BillyONeal thanks and apologies for any delays. Was my first port :)

**Pull request merged**